lib/sierra-models/base.js

```diff
@@ -156,24 +156,74 @@ class SierraBase {
     return matches
   }
 
+  /**
+   *  Given an array of VarFieldMatch objects, returns a new array of matches
+   *  with dupes removed. A dupe entry is:
+   *   - An entry where there's another entry with an identical primary and
+   *     parallel value
+   *   - Orphaned parallel where there exists another entry with an identical
+   *     parallel that also has a primary value
+   *   - Orphaned primary where there exists another entry with an identical
+   *     primary that also has a parallel
+   */
   _uniqueVarFieldMatches (matches) {
     // Create a hash function, which will be used to determine equality of two
     // different maches. We consider two VarFieldMatches as being equal if
     // their primary and parallel values are equal:
     const hasher = (entry) => [entry.value, entry.parallel?.value].join('🎸')
+    let unique = uniqueObjectsByHash(matches, hasher)
 
-    const unique = uniqueObjectsByHash(matches, hasher)
+    // Also remove any orphans whose value matches non-orphans:
+    unique = this._removeRedundantOrphans(unique)
 
     if (unique.length < matches.length) {
-      const dupes = dupeObjectsByHash(matches, hasher)
-        // Only show first entry in each set of dupes, since they're presumably identical:
-        .map((set) => set[0])
-      logger.debug(`Removed ${matches.length - unique.length} dupe VarFieldMatch entries`, dupes)
+      logger.debug(`Removed ${matches.length - unique.length} dupe VarFieldMatch entries`)
     }
 
     return unique
   }
 
+  /**
+   *  Given an array of VarFieldMatch objects, removes any that are redundant
+   *  given the other entries. A match is considered redundant if:
+   *   - It's an orphaned parallel and there exists another entry with an
+   *     identical parallel that also has a primary value, or:
+   *   - It's an orphaned primary and there exists another entry with an
+   *     identical primary that also has a parallel, or:
+   *
+   *  For example, given this set of entries:
+   *   - value: v1
+   *     parallel:
+   *       value: v1 parallel
+   *   - parallel:
+   *       value: v1 parallel
+   *
+   *  .. This function should remove the second entry because it's redundant given the first entry.
+   */
+  _removeRedundantOrphans (matches) {
+    return matches.reduce((a, match) => {
+      // If it's an orphaned parallel
+      const isOrphanedParallel = !!(!match.value && match.parallel?.value)
+      const isOrphanedPrimary = !!(match.value && !match.parallel?.value)
+      const isRedundantOrphan = (isOrphanedParallel || isOrphanedPrimary) && !!matches.find((other) => {
```

Comment:
I like the semantic isOrphaned* variables. Can you create a similar isRedundant using this matches.find logic here?

Reply:
But `isRedundant` depends on the VarFieldMatch entity being an orphan; A VarFieldMatch entity has to be an orphan in order for it to be "redundant". I feel that setting up a `isRedundant` var would be misleading because it would sometimes be `true` for things that VarFieldMatches that are not in fact redundant. Unless I restructured to this:

```
const isOrphanedParallel = !!(!match.value && match.parallel?.value)
const isOrphanedPrimary = !!(match.value && !match.parallel?.value)
const isRedundantOrphan = !!matches.find((other) => { 
  // If an identical parallel value is associated with a primary value, remove the orphaned parallel:
  if (isOrphanedParallel) { 
    return other.value && other.parallel?.value === match.parallel.value

  // If an identical primary value is associated with a parallel value, the orphaned primary is redundant:
  } else if (isOrphanedPrimary) { 
    return other.parallel?.value && other.value === match.value
  } 
})
```

It wastes some cycles because we're running that `find` for _all_ VarFieldMatch entities - not just those that have already been identified as either an orphaned parallel or primary. But maybe easier to follow?

Reply:
ah ok. makes sense to leave it as is.